client/app/startup/SearchPageApp.js

```diff
@@ -27,18 +28,36 @@ export default (props) => {
 
   const bootstrappedData = TransitImmutableConverter.fromJSON(props.data);
 
-  const listingsToMap = (listings) =>
+  const listingWithAuthor = (listing, profiles) => {
+    const author = profiles.get(listing.authorId);
+    return listing.set('author', author);
+  };
+
+  const profilesToMap = (includes) =>
+    includes.reduce((acc, val) => {
+      const type = val.get(':type');
+      if (type === ':profile') {
+        const profile = parseProfile(val);
+        const id = val.get(':id');
+        return acc.set(id, profile);
+      } else {
+        return acc;
+      }
+    }, new Immutable.Map());
```

Comment:
I'm starting to think we need to plan some kind of structure for these conversions.